File: sdk/php/gui/controlsfx/UXPopOver.php
<?php
namespace php\gui\controlsfx;

use php\gui\UXPopupWindow;
use php\gui\UXNode;

class UXPopOver extends UXPopupWindow{
    /**
     * @var string
    */
    public $title;
    /**
     * @var boolean
    */
    public $detachable;
    /**
     * @var boolean
    */
    public $animated;
    /**
     * @var STRING
     * BOTTOM_CENTER, BOTTOM_LEFT, BOTTOM_RIGHT, LEFT_BOTTOM, LEFT_CENTER, LEFT_TOP, RIGHT_BOTTOM, RIGHT_CENTER, RIGHT_TOP, TOP_CENTER, TOP_LEFT, TOP_RIGHT 
    */
    public $arrowLocation;
    /**
     * @var int
    */
    public $fadeInDuration;
    /**
     * @var int
    */
    public $fadeOutDuration;
    /**
     * @var UXNode
    */
    public $contentNode;
    /**
     * @var boolean
    */
    public $closeButtonEnabled;
    /**
     * @var boolean
    */
    public $detached;
    /**
     * @var double
    */
    public $arrowSize;
    /**
     * @var double
    */
    public $arrowIndent;
    /**
     * @var double
    */
    public $cornerRadius;
    /**
     * @var boolean
    */
    public $headerAlwaysVisible;


    public function hideWithFade(int $duration){
        
    }
}

File: src-jvm/main/java/org/develnext/jphp/ext/controlsfx/classes/control/UXPopOver.java
package org.develnext.jphp.ext.controlsfx.classes.control;

import javafx.geometry.Side;
import javafx.scene.Node;
import javafx.stage.Window;
import javafx.util.Duration;
import org.controlsfx.control.MasterDetailPane;
import org.controlsfx.control.PopOver;
import org.develnext.jphp.ext.controlsfx.ControlsFXExtension;
import org.develnext.jphp.ext.javafx.classes.UXControl;
import org.develnext.jphp.ext.javafx.classes.UXPopupWindow;
import php.runtime.annotation.Reflection;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;



@Reflection.Namespace(ControlsFXExtension.NS)
@Reflection.Name("UXPopOver")
public class UXPopOver extends UXPopupWindow<PopOver> {
    public UXPopOver(Environment env, PopOver wrappedObject) {
        super(env, wrappedObject);
    }
    public UXPopOver(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    interface WrappedInterface{
        @Reflection.Property String title();

        @Reflection.Property boolean animated();
        @Reflection.Property boolean closeButtonEnabled();
        @Reflection.Property boolean detachable();
        @Reflection.Property boolean detached();
        @Reflection.Property boolean headerAlwaysVisible();

        @Reflection.Property double arrowIndent();
        @Reflection.Property double arrowSize();
        @Reflection.Property double cornerRadius();

        @Reflection.Property Duration fadeInDuration();
        @Reflection.Property Duration fadeOutDuration();

        @Reflection.Property PopOver.ArrowLocation arrowLocation();
        @Reflection.Property Node contentNode();
    }

    @Reflection.Signature
    public void __construct(){
        __wrappedObject = new PopOver();
    }
    @Reflection.Signature
    public void showByNode(Node owner){
        getWrappedObject().show(owner);
    }
    @Reflection.Signature
    public void showByNode(Node owner, double offset){
        getWrappedObject().show(owner, offset);
    }
    @Reflection.Signature
    public void showByNode(Node owner, double x, double y){
        getWrappedObject().show(owner, x, y);
    }
    @Reflection
    public void showByNode(Node owner, double x, double y, Duration fadeInDuration){
        getWrappedObject().show(owner, x, y, fadeInDuration);
    }
    @Reflection.Signature
    public void show(Window owner, double x, double y){
        getWrappedObject().show(owner, x, y);
    }
    @Reflection.Signature
    public void show(Window owner){
        getWrappedObject().show(owner);
    }

    @Reflection.Signature
    public void hideWithFade(Duration duration){
        getWrappedObject().hide(duration);
    }
}
